Support upgrades and consul service registration

rgevaert opened the pull request:
This commits add the changes we made from the jsok repo.

I can understand this will need some work on how to get it included. We would really like to switch to a voxpupuli module and we are expecting now some movement and support from the community.

bastelfreak commented:
thanks for the PR. I will have a look as soon as I finished https://github.com/voxpupuli/puppet-vault/pull/1 (which will configure CI and apply vox pupuli defaults).

rgevaert commented:
Thanks. Totally understand, agree and support doing #1 first. This will probably drop two commits from the PR.

bastelfreak commented:
@rgevaert done. Can you please rebase/drop the pdk update commit?

rgevaert commented:
@bastelfreak sorry for the delay but other things came in between and this isn't pressing for us as we have a working setup. Anyhow I removed the commit and fixed lint and rubocop issues.

rgevaert commented on manifests/params.pp:
```diff
@@ -4,6 +4,59 @@
 # @api private
 #
 class vault::params {
+  $user               = 'vault'
```
@bastelfreak sorry for that. I wasn't aware. It's been ages since I did some puppet development actually... Never got on the voxpupuli train, and now I am not really doing real puppet module development.

Nevertheless, to double check, you are not only talking about just the $user variable? But about all other static ones too. Right?

bastelfreak replied:
hey, no problem. yes, please move all static data. we also have some review guidelines at https://voxpupuli.org/docs/reviewing_pr/

jonasdemoor commented:
Hi, how can I re-trigger the acceptance tests with GitHub Actions? There were some issues with apt.releases.hashicorp.com at the time of testing. The tests seem to work now on my local machine. 

Output snippet from the failed test run at the time of testing:
```
Notice: /Stage[main]/Apt::Update/Exec[apt_update]/returns: Ign:1 https://apt.releases.hashicorp.com/ bionic InRelease
  Notice: /Stage[main]/Apt::Update/Exec[apt_update]/returns: Hit:2 http://security.ubuntu.com/ubuntu bionic-security InRelease
  Notice: /Stage[main]/Apt::Update/Exec[apt_update]/returns: Hit:3 http://archive.ubuntu.com/ubuntu bionic InRelease
  Notice: /Stage[main]/Apt::Update/Exec[apt_update]/returns: Err:4 https://apt.releases.hashicorp.com/ bionic Release
  Notice: /Stage[main]/Apt::Update/Exec[apt_update]/returns:   404  Not Found [IP: 13.225.41.31 443]
  Notice: /Stage[main]/Apt::Update/Exec[apt_update]/returns: Hit:5 http://apt.puppetlabs.com/ bionic InRelease
  Notice: /Stage[main]/Apt::Update/Exec[apt_update]/returns: Hit:6 http://archive.ubuntu.com/ubuntu bionic-updates InRelease
  Notice: /Stage[main]/Apt::Update/Exec[apt_update]/returns: Hit:7 http://archive.ubuntu.com/ubuntu bionic-backports InRelease
  Notice: /Stage[main]/Apt::Update/Exec[apt_update]/returns: Reading package lists...
  Notice: /Stage[main]/Apt::Update/Exec[apt_update]/returns: E: The repository 'https://apt.releases.hashicorp.com/ bionic Release' does not have a Release file.
  Error: /Stage[main]/Apt::Update/Exec[apt_update]: Failed to call refresh: '/usr/bin/apt-get update' returned 100 instead of one of [0]
  Error: /Stage[main]/Apt::Update/Exec[apt_update]: '/usr/bin/apt-get update' returned 100 instead of one of [0]
  Error: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install vault' returned 100: Reading package lists...
  Building dependency tree...
  Reading state information...
  E: Unable to locate package vault
  Error: /Stage[main]/Vault::Install/Package[vault]/ensure: change from 'purged' to 'present' failed: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install vault' returned 100: Reading package lists...
  Building dependency tree...
  Reading state information...
  E: Unable to locate package vault
  ```

bastelfreak commented:
hey @jonasdemoor. You can restart the pipeline be closing and reopening the PR. I wanted to do that and accidently merged it. I reverted that in #8. Can you please recreate the PR? Sorry for the inconvinience.

jonasdemoor commented:
Hi @bastelfreak, no worries :) Thanks for looking into this, I opened a new PR .
